thunderbird-todoist/scripts/common.js

```diff
@@ -26,14 +34,51 @@ function fillAllProjectsSelect(selectid) {
             process(child, indent + 1);
           });
         }
-        const el = document.getElementById(selectid);
+        const el = document.getElementById(projectselect);
         el.innerHTML = "";
         projects.forEach((proj) => {
           process(proj, 0);
         });
       })
       .catch((err) => {
-        const el = document.getElementById(selectid);
+        const el = document.getElementById(projectselect);
+        el.innerHTML =
+          '<option value="0">Could not connect to Todoist...</option>';
+      });
+  });
+}
+
+function fillAllCollaboratorsSelect(collaboratorselect, projectid) {
+  loadDefaultCollaborator().then((selected) => {
+    getAllCollaborators(projectid)
+      .then((collaborators) => {
+        function process(collaborator, indent) {
+          let option = document.createElement("option");
+          let text = "";
+          for (let i = 0; i < indent; i++) {
+            text += "&nbsp;&nbsp;&nbsp;";
+          }
+          option.innerHTML = text + collaborator.name;
+          option.value = collaborator.id;
+          if (collaborator.id == selected) {
+            option.selected = true;
+          }
+          el.add(option);
+        }
+        const el = document.getElementById(collaboratorselect);
+        el.innerHTML = "";
+        let arrayEmpty = true;
+        Object.values(collaborators).forEach((collaborator) => {
+          process(collaborator, 0);
+          arrayEmpty = false;
+        });
+        if (arrayEmpty) {
+          el.innerHTML =
+          '<option value="0">No collaborators in this project</option>';  
```

Comment:
In this case it might be better to completely hide the `select` or to reduce its height. At least for me, most of my projects don't have any collaborators, and I would find it a bit annoying if the list was always displayed just to tell me that there are no collaborators. On the other hand some people might be confused if this is only displayed for some projects... Not sure how this should be done best, but probably worth it to think about a bit more.